examples/variational_autoencoder.py

```diff
@@ -20,15 +20,15 @@
 epsilon_std = 1.0
 
 
-x = Input(batch_shape=(batch_size, original_dim))
+x = Input(shape=(original_dim,))
 h = Dense(intermediate_dim, activation='relu')(x)
 z_mean = Dense(latent_dim)(h)
 z_log_var = Dense(latent_dim)(h)
 
 
 def sampling(args):
     z_mean, z_log_var = args
-    epsilon = K.random_normal(shape=(batch_size, latent_dim), mean=0.,
+    epsilon = K.random_normal(shape=(K.shape(z_mean)[0], latent_dim), mean=0.,
```

Comment:
How so? I use TF and it didn't crash on my machine, this is the output of variational_autoencoder.py on the two backends.
Theano output up to the end of first epoch (I don't know if it's running on GPU because I never use it):
```
Using Theano backend.
BATCH SIZE: 51
variational_autoencoder.py:69: UserWarning: Output "custom_variational_layer_1" missing from loss dictionary. We assume this was done on purpose, and we will not be expecting any data to be passed to "custom_variational_layer_1" during training.
  vae.compile(optimizer='rmsprop', loss=None)
Train on 60000 samples, validate on 10000 samples
Epoch 1/50
60000/60000 [==============================] - 12s - loss: 180.6366 - val_loss: 168.7532
```
Tensorflow output up to the end of first epoch :
```
Using TensorFlow backend.
BATCH SIZE: 51
variational_autoencoder.py:69: UserWarning: Output "custom_variational_layer_1" missing from loss dictionary. We assume this was done on purpose, and we will not be expecting any data to be passed to "custom_variational_layer_1" during training.
  vae.compile(optimizer='rmsprop', loss=None)
Train on 60000 samples, validate on 10000 samples
Epoch 1/50
2017-08-22 19:38:49.520646: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:893] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2017-08-22 19:38:49.520919: I tensorflow/core/common_runtime/gpu/gpu_device.cc:955] Found device 0 with properties: 
name: GeForce GTX 1050
major: 6 minor: 1 memoryClockRate (GHz) 1.493
pciBusID 0000:01:00.0
Total memory: 3.95GiB
Free memory: 3.64GiB
2017-08-22 19:38:49.520948: I tensorflow/core/common_runtime/gpu/gpu_device.cc:976] DMA: 0 
2017-08-22 19:38:49.520953: I tensorflow/core/common_runtime/gpu/gpu_device.cc:986] 0:   Y 
2017-08-22 19:38:49.520977: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1045] Creating TensorFlow device (/gpu:0) -> (device: 0, name: GeForce GTX 1050, pci bus id: 0000:01:00.0)
60000/60000 [==============================] - 3s - loss: 181.1762 - val_loss: 169.2895
```